mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistrationHandler.java

```diff
@@ -49,9 +52,12 @@
         this.groupRegistrations = new ConcurrentHashMap<>();
     }
 
-    GroupRegistration retrieveGroupRegistration(Group group) {
-        return Optional.ofNullable(groupRegistrations.get(group))
-            .orElseThrow(() -> new GroupRegistrationNotFound(group));
+    Publisher<GroupRegistration> retrieveGroupRegistration(Group group) {
```

Comment:
We don't need a publisher for an implementation class, it makes sense only in API

Reply:
I reverted, as your suggestion to use the EventDispatcher